src/utils/subscription-registry.js

```diff
@@ -260,41 +257,41 @@ class SubscriptionRegistry {
    * @returns {void}
    */
   subscribe(name, socketWrapper) {
-    if (this._subscriptions[name] === undefined) {
-      this._subscriptions[name] = []
+    let sockets = this._subscriptions.get(name)
+
+    if (!sockets) {
+      this._subscriptions.set(name, sockets = [])
     }
 
-    if (this._subscriptions[name].indexOf(socketWrapper) !== -1) {
+    const index = utils.sortedIndexBy(sockets, socketWrapper, 'uuid')
+
+    if (sockets[index] === socketWrapper) {
       const msg = `repeat supscription to "${name}" by ${socketWrapper.user}`
       this._options.logger.log(C.LOG_LEVEL.WARN, this._constants.MULTIPLE_SUBSCRIPTIONS, msg)
       socketWrapper.sendError(this._topic, this._constants.MULTIPLE_SUBSCRIPTIONS, name)
       return
     }
 
-    if (!this.isLocalSubscriber(socketWrapper)) {
-      const unsubscribeAllFn = this.unsubscribeAll.bind(this, socketWrapper)
-      this._unsubscribeAllFunctions.push({
-        socketWrapper,
-        fn: unsubscribeAllFn
+    sockets.splice(index, 0, socketWrapper)
+
+    let names = this._sockets.get(socketWrapper)
```

Comment:
`names` could be stored as a magic property on `socketWrapper`. However, I'm not sure that kind of "hack" is worth the potential performance improvement.